torch/autograd/function.py

```diff
@@ -22,6 +22,13 @@ def save_for_backward(self, *tensors: torch.Tensor):
         incorrect gradients and memory leaks, and enable the application of saved
         tensor hooks. See :class:`torch.autograd.graph.saved_tensors_hooks`.
 
+        Note that if tensors that are neither input nor output are saved for backward,
+        your custom Function may not support
+        `double backward <https://pytorch.org/tutorials/intermediate/custom_function_double_backward_tutorial.html>`_.
+        Custom Functions that do not support double backward should decorate their
+        :func:`backward` method with `@once_differentiable` so that performing
+        double backward raises an error.
```

Comment:
nit do you want to add one more sentence here that if you need double backward you should either:
- Recompute that intermediary based on the inputs during the backward
- Make this intermediary an output of the custom Function